cogs/servermanage/servermanage.py

```diff
@@ -0,0 +1,439 @@
+"""Server Manage Cog, to help manage server icon and banners."""
+import logging
+import time
+import asyncio
+from datetime import date, datetime, timedelta
+from os.path import splitext
+from os import remove
+from pathlib import Path
+import discord
+from redbot.core import Config, checks, commands, data_manager
+from redbot.core.commands.context import Context
+from redbot.core.utils import AsyncIter
+from redbot.core.utils.chat_formatting import error, pagify, warning
+from redbot.core.utils.menus import DEFAULT_CONTROLS, menu
+from redbot.core.bot import Red
+
+from .commands import ServerManageCommands
+from .exceptions import InvalidAttachmentsError, InvalidFileError, InvalidImageError
+from .meta import ServerManageMeta
+
+BASE_GUILD = {"icons": {}, "iconsDates": {}, "banners": {}, "bannersDates": {}}
+
+
+class ServerManage(ServerManageCommands, commands.Cog, metaclass=ServerManageMeta):
+    """Auto-assign server banner and icon on configurable days."""
+
+    # Class constructor
+    def __init__(self, bot: Red):
+        self.bot = bot
+        self.config = Config.get_conf(self, identifier=5842647, force_registration=True)
+        # Register default (empty) settings.
+        self.config.register_guild(**BASE_GUILD)
+
+        # Initialize logger, and save to cog folder.
+        self.dataFolder = data_manager.cog_data_path(cog_instance=self)
+        self.logger = logging.getLogger("red.luicogs.ServerManage")
+        if self.logger.level == 0:
+            # Prevents the self.logger from being loaded again in case of module reload.
+            self.logger.setLevel(logging.INFO)
+            handler = logging.FileHandler(
+                filename=str(self.dataFolder) + "/info.log", encoding="utf-8", mode="a"
+            )
+            handler.setFormatter(
+                logging.Formatter("%(asctime)s %(message)s", datefmt="[%d/%m/%Y %H:%M:%S]")
+            )
+            self.logger.addHandler(handler)
+
+        # On cog load, we want the loop to run once.
+        self.lastChecked = datetime.now() - timedelta(days=1)
+        self.bgTask = self.bot.loop.create_task(self.imageLoop())
+
+    # Cancel the background task on cog unload.
+    def __unload(self):
+        self.bgTask.cancel()
+
+    def cog_unload(self):
+        self.__unload()
+
+    async def imageLoop(self):
+        while True:
+            if self.lastChecked.day != datetime.now().day:
+                self.logger.info("Checking to see if we need to change server images")
+                self.lastChecked = datetime.now()
+                for guild in self.bot.guilds:
+                    await self.checkGuildIcons(guild)
+                    await self.checkGuildBanners(guild)
+            await asyncio.sleep(60)
+
+    async def checkGuildIcons(self, guild: discord.Guild):
+        self.logger.debug("Checking guild icon for %s (%s)", guild.name, guild.id)
```

Comment:
Do we want to leave in this debug as it will be login it every minute?

Reply:
This will happen once a day if the DEBUG level is enabled, since it's called only when the day changes. By default it will be at the INFO level, so this trace will not show up.

Reply:
👍